tests/Aspire.Workload.Tests/WorkloadTestsBase.cs

```diff
@@ -0,0 +1,263 @@
+        HashSet<string> foundNames = [];
+        List<ResourceRow> foundRows = [];
+
+        var timeout = TimeSpan.FromSeconds(timeoutSecs);
+        Stopwatch sw = new();
+        sw.Start();
+
+        while (foundNames.Count < expectedRowsTable.Count && sw.Elapsed < timeout)
+        {
+            await Task.Delay(500);
+
+            // _testOutput.WriteLine($"Checking for rows again");
+            ILocator rowsLocator = dashboardPage.Locator("//fluent-data-grid-row[@class='resource-row']");
+            var allRows = await rowsLocator.AllAsync();
+            // _testOutput.WriteLine($"found rows#: {allRows.Count}");
+            if (allRows.Count == 0)
+            {
+                // Console.WriteLine ($"** no rows found ** elapsed: {sw.Elapsed.TotalSeconds} secs");
+                continue;
+            }
+
+            foreach (ILocator rowLoc in allRows)
+            {
+                // get the cells
+                IReadOnlyList<ILocator> cellLocs = await rowLoc.Locator("//fluent-data-grid-cell[@role='gridcell']").AllAsync();
+                Assert.Equal(8, cellLocs.Count);
+
+                // is the resource name expected?
+                string resourceName = await cellLocs[1].InnerTextAsync();
+                if (!expectedRowsTable.TryGetValue(resourceName, out var expectedRow))
+                {
+                    Assert.Fail($"Row with unknown name found: {resourceName}");
+                }
+                if (foundNames.Contains(resourceName))
+                {
+                    continue;
+                }
+
+                string resourceNameInCell = await cellLocs[1].InnerTextAsync().ConfigureAwait(false);
+                resourceNameInCell.Trim();
+                AssertEqual(expectedRow.Name, resourceNameInCell, $"Name for {resourceName}");
+
+                string actualState = await cellLocs[2].InnerTextAsync().ConfigureAwait(false);
+                actualState = actualState.Trim();
+                if (expectedRow.State != actualState && actualState != "Finished" && !actualState.Contains("failed", StringComparison.OrdinalIgnoreCase))
+                {
+                    testOutput.WriteLine($"[{expectedRow.Name}] expected state: '{expectedRow.State}', actual state: '{actualState}'");
+                    continue;
+                }
+                AssertEqual(expectedRow.State, await cellLocs[2].InnerTextAsync(), $"State for {resourceName}");
+
+                // Match endpoints
+
+                int matchingEndpoints = 0;
+                var expectedEndpoints = expectedRow.Endpoints;
+
+                string[] endpointsFound =
+                    (await rowLoc.Locator("//div[@class='fluent-overflow-item']").AllAsync())
+                        .Select(async e => await e.InnerTextAsync())
+                        .Select(t => t.Result.Trim(','))
+                        .ToArray();
+                // FIXME: this could still return "+2"
```

Comment:
What's the plan to address this?

Reply:
um I don't have any immediate plan on fixing this. I haven't hit this yet, but it is a possibility. And the comment might help somebody 🤷